include/onnxruntime/core/common/string_utils.h

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// Portions Copyright (c) Microsoft Corporation
+
+#pragma once
+
+#include <locale>
+#include <sstream>
+#include <type_traits>
+
+namespace onnxruntime {
+
+namespace detail {
+inline void MakeStringImpl(std::ostringstream& /*ss*/) noexcept {
+}
+
+template <typename T>
+inline void MakeStringImpl(std::ostringstream& ss, const T& t) noexcept {
+  ss << t;
+}
+
+template <typename T, typename... Args>
+inline void MakeStringImpl(std::ostringstream& ss, const T& t, const Args&... args) noexcept {
+  MakeStringImpl(ss, t);
+  MakeStringImpl(ss, args...);
+}
+}  // namespace detail
+
+/**
+ * Makes a string by concatenating string representations of the arguments.
+ */
+template <typename... Args>
+std::string MakeString(const Args&... args) {
+  std::ostringstream ss;
+  ss.imbue(std::locale::classic());
+  detail::MakeStringImpl(ss, args...);
+  return ss.str();
+}
+
+// MakeString versions for already-a-string types.
+
+inline std::string MakeString(const std::string& str) {
+  return str;
+}
+
+inline std::string MakeString(const char* cstr) {
+  return cstr;
+}
+
+/**
+ * Tries to parse a value from an entire string.
+ */
+template <typename T>
+bool TryParse(const std::string& str, T& value) {
```

Comment:
Would using stol family of functions be simpler?

Reply:
not sure if it would. one nice thing about operator>> is that i don't need to call different functions explicitly. also can specify a particular locale without having to update the global one.

---
In reply to: [535837295](https://github.com/microsoft/onnxruntime/pull/5923#discussion_r535837295) [](ancestors = 535837295)